### PR TITLE
ntp: use openssl@3

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
```
  In file included from /usr/include/pthread.h:33,
                   from work_thread.c:13:
  work_thread.c:45:57: error: missing binary operator before token "("
     45 | #if defined(PTHREAD_STACK_MIN) && THREAD_MINSTACKSIZE < PTHREAD_STACK_MIN
        |                                                         ^~~~~~~~~~~~~~~~~
```

**Review by SMillerDev — APPROVED** [review]

File: Formula/ntp.rb [hunk 1]
```diff
@@ -22,20 +22,25 @@ class Ntp < Formula
...
-  depends_on "openssl@1.1"
+  depends_on "openssl@3"
+
+  # Patch for glibc 2.34+ compatibility
+  patch do
+    on_linux do
+      url "https://raw.githubusercontent.com/Homebrew/formula-patches/6c508ba/ntp/ntp-4.2.8p15-glibc.patch"
+      sha256 "f41a8066ce41f3f1d066801f54ead54b66d13e71355a50c20836d303f8b300e4"
+    end
+  end
```
Comment (SMillerDev) [inline Formula/ntp.rb:33]: Is this upstreamed?

**Review by SMillerDev — CHANGES_REQUESTED** [review]

File: Formula/ntp.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/ntp.rb:33]: I did not find a place to submit this patch or see this patch shown up in the upstream.

In fact, according to https://lore.kernel.org/all/20210923233409.50cba2f7@gmx.net/T/

 >> Yeah, it looks like ntp is a pretty dead-end project...

 > Maybe time to provide NTPsec [1] as alternative...

 > Regards,
 > Peter

 > [1] https://www.ntpsec.org/

Comment (chenrui333) [reply Formula/ntp.rb:33]: Maybe we can simply deprecate the formulae in this same PR?

**Review by iMichka — CHANGES_REQUESTED** [review]
@chenrui333 let's deprecate this then.

**Comment by p-linnane** [comment]
https://github.com/Homebrew/homebrew-core/pull/119175

**Pull request closed without merging**